graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -584,6 +628,305 @@ def prune_output_blocks_in_compound_match_query(compound_match_query):
+
+    Args:
+        expression: BinaryComposition without any ContextField operand(s)
+
+    Returns:
+        simplified expression if the given expression is a disjunction/conjunction
+        and one of it's operands is a TrueLiteral,
+        and the original expression otherwise
+    """
+    if expression.operator == u'||':
+        if expression.left == TrueLiteral or expression.right == TrueLiteral:
+            return TrueLiteral
+        else:
+            return expression
+    elif expression.operator == u'&&':
+        if expression.left == TrueLiteral:
+            return expression.right
+        if expression.right == TrueLiteral:
+            return expression.left
+        else:
+            return expression
+    else:
+        return expression
+
+
+def _simplify_ternary_conditional(expression):
+    """Return the `if_true` clause if the predicate of the TernaryConditional is a TrueLiteral.
+
+    Args:
+        expression: TernaryConditional to be simplified.
+
+    Returns:
+        simplified TernaryConditional, if the predicate is True,
+        and the original expression otherwise
+    """
+    if expression.predicate == TrueLiteral:
+        return expression.if_true
+    else:
+        return expression
+
+
+def _update_context_field_expression(expression, present_locations):
+    """Lower Expressions involving non-existent ContextFields True, and simplify result."""
+    no_op_blocks = (ContextField, Literal, LocalField, UnaryTransformation, Variable)
+    if isinstance(expression, BinaryComposition):
+        if isinstance(expression.left, ContextField) or isinstance(expression.right, ContextField):
+            return _update_context_field_binary_composition(expression, present_locations)
+        else:
+            return _simplify_non_context_field_binary_composition(expression)
+    elif isinstance(expression, TernaryConditional):
+        return _simplify_ternary_conditional(expression)
+    elif isinstance(expression, BetweenClause):
+        lower_bound = expression.lower_bound
+        upper_bound = expression.upper_bound
+        if isinstance(lower_bound, ContextField) or isinstance(upper_bound, ContextField):
+            raise AssertionError(u'Found BetweenClause with ContextFields as lower/upper bounds. '
+                                 u'This should never happen: {}'.format(expression))
+    elif isinstance(expression, (OutputContextField, FoldedOutputContextField)):
+        raise AssertionError(u'Found unexpected expression of type {}. This should never happen: '
+                             u'{}'.format(type(expression).__name__, expression))
+    elif isinstance(expression, no_op_blocks):
+        return expression
+    else:
+        raise AssertionError(u'Found unexpected expression of type {}. This should never happen: '
+                             u'{}'.format(type(expression).__name__, expression))
+
+
+def _construct_update_context_field_visitor_fn(present_locations):
```

Comment:
I think I had a comment on this in a previous iteration that didn't get addressed. I would do two things here:
- make `_update_context_field_expression` take the `present_locations` as the first argument, since it's the one that is less likely to change
- then, this entire function is replaceable by just `partial(_update_context_field_expression, present_locations)` so you can just put that code into the place where this function is invoked